sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -86,6 +86,7 @@ ssize_t gnrc_sock_recv(gnrc_sock_reg_t *reg, gnrc_pktsnip_t **pkt_out,
             if (msg.content.value == _TIMEOUT_MAGIC) {
                 return -ETIMEDOUT;
             }
+            __attribute__((fallthrough));
```

Comment:
> but in your provided examples in #7403 (comment) you didn't try to put the comment above of the case.

Ah, I see what you mean. It falls into the default as seen below. The lowest one in the example above is above the default case, only seperated by `#endif`
```
        case GNRC_NETAPI_MSG_TYPE_RCV:
            pkt = msg.content.ptr;
            break;
#ifdef MODULE_XTIMER
        case _TIMEOUT_MSG_TYPE:
            if (msg.content.value == _TIMEOUT_MAGIC) {
                return -ETIMEDOUT;
            }   
            /* Attribute instead of "Falls Through.", because 
             * comments don't work in #ifdef */
            __attribute__((fallthrough));
#endif
        default:
```